src/Illuminate/Contracts/Routing/UrlGenerator.php

```diff
@@ -19,6 +19,22 @@ public function current();
      */
     public function previous($fallback = false);
 
+    /**
+     * Get the URL for the previous request without the query string.
+     *
+     * @param  mixed  $fallback
+     * @return string
+     */
+    public function previousWithoutQuery($fallback = false);
```

Comment:
adding methods to a non-internal interface is a major breaking change

Reply:
Oh.. I'm unsure now, but isn't the `UrlGenerator` interface internal?

Reply:
no. it's in the contracts repo. and not marked with `@internal`

Reply:
Do I understand correctly that other projects might use `laravel/framework` and thus the `UrlGenerator` contract have induce breaking changes?

Reply:
Could we simply add `@internal This method is not meant to be used or overwritten outside the framework.` for `previousWithoutQuery` and `previousPath` then? 

Or would it be better to simply not add these methods to the contract?

Reply:
@miclaus It doesn't matter if the functions are marked with `@internal` because anyone who implements this interface will still have to add those two functions to their own implementation. If they don't their app will break.

Reply:
@dennisprudlo Got it now, thanks! I removed the methods from the contract.